lib/imagec/docker.go

```diff
@@ -146,18 +149,22 @@ func LearnAuthURL(options ImageCOptions) (*url.URL, error) {
 }
 
 // FetchToken fetches the OAuth token from OAuth endpoint
-func FetchToken(url *url.URL) (*Token, error) {
+func FetchToken(ic *ImageC, url *url.URL, progressOutput progress.Output) (*Token, error) {
```

Comment:
Actually while at it, we could add ctx to the LearnX ones and use timeouts on them too